common/log/loggerimpl/logger.go

```diff
@@ -62,9 +61,17 @@ func NewDevelopment() (log.Logger, error) {
 
 // NewLogger returns a new logger
 func NewLogger(zapLogger *zap.Logger) log.Logger {
+	return NewLoggerWithSampleFunc(zapLogger, func(i int) bool {
+		return rand.Intn(i) == 0
+	})
+}
+
+// NewLoggerWithSampleFunc returns a new logger with sample function.
+func NewLoggerWithSampleFunc(zapLogger *zap.Logger, sampleLocalFn func(int) bool) log.Logger {
```

Comment:
new func means new code to cover. instead of creating separate constructor we could pass optional params to existing one. that would require making `loggerImpl` public though. up to you
```
type LoggerOption func(*LoggerImpl)

func WithSamplingFn(fn func(int) bool) LoggerOption {
  return func(l *LoggerImpl) {
    l.sampleLocalFn = fn
 }
}

func NewLogger(zapLogger *zap.Logger, opts ...LoggerOption) log.Logger {
  l := &LoggerImpl{...}
  for _, o := range opts {
     o(l)
  }
  return l
}
```

Reply:
Good call. Though it is not required to make LoggerImpl public. It is usually the case when you want to limit which options are provided. I've switched to options and, since we are talking about tests, reworked tests and improved coverage.